apps/vaporgui/AppSettingsMenu.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <QDialog>
+#include "PWidget.h"
+#include "PFileSelectorHLI.h"
+#include "VFileSelector.h"
+
+class QWidget;
+class SettingsParams;
+class PGroup;
+class PSection;
+
+//! \class Preferences Menu
+//! \ingroup Public_GUI
+//! \brief Menu for global application preferences
+//! \author Scott Pearse
+
+class AppSettingsMenu : public QDialog {
+    Q_OBJECT
+
+public:
+    AppSettingsMenu(QWidget *parent);
+    void Update(SettingsParams *sp);
```

Comment:
There is an Updatable interface to standardize this.

Reply:
I'm not sure what you're referring to.  We don't currently have a QDialog based PWidget, and our current dialogs (ParamsWidgetDemo, PythonVariables, Statistics, RenderHolder, etc) implement the Update() function it on their own.

Do you mean that this should be inheriting from PWidget, and use its Update() function?

Reply:
There is an `Updatable` class. All the other dialogs you mention were implemented before.

Reply:
Fixed.